### PR TITLE
feat: add sale end year filter

### PR DESCRIPTION
**Pull request by egdbear**
The type of this PR is: **Feat**

This PR solves [DIA-99]

### Description

Adds Sale Year filter for Auction Results page.

cc @artsy/diamond-devs 
image

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DIA-99]: https://artsyproduct.atlassian.net/browse/DIA-99?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/Artist/Routes/AuctionResults/Components/AuctionFilters/SaleEndYearFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  const options = (
+    aggregations
+      ?.find(aggregation => aggregation.slice === "LOTS_BY_SALE_YEAR")
+      ?.counts.filter(c => c !== null) || []
+  ).map(c => ({
+    text: c?.name,
+    value: c?.name,
+  }))
+
+  React.useEffect(() => {
```
Comment (egdbear) [inline src/Apps/Artist/Routes/AuctionResults/Components/AuctionFilters/SaleEndYearFilter.tsx:27]: Not sure if this makes sense here, I'm looking for _suggestions_  😄

Comment (dzucconi) [reply src/Apps/Artist/Routes/AuctionResults/Components/AuctionFilters/SaleEndYearFilter.tsx:27]: Why can't this logic happen on the select callbacks?

Comment (egdbear) [reply src/Apps/Artist/Routes/AuctionResults/Components/AuctionFilters/SaleEndYearFilter.tsx:27]: Do you have an example? 

```
          onSelect={year =>
                saleStartYear === null && options.length > 0
                  ? setFilter?.("saleStartYear", parseInt(options[0].value))
                  : setFilter?.("saleStartYear", parseInt(year))
              }
```

This won't work as expected :|

Comment (dzucconi) [reply src/Apps/Artist/Routes/AuctionResults/Components/AuctionFilters/SaleEndYearFilter.tsx:27]: I suppose I don't follow exactly what this is doing without pulling it; I can take a look in a bit. If the effect works that's fine with me but they are much harder to reason about and frequently introduce bugs when performing something directly in callbacks is more clear.

**Comment by dzucconi** [comment]
OK — so I checked this out.

We need some clarification on the behavior here. If I select an end year; presumably the start year should only show years UP TO that year, right?

I think we should actively prevent invalid inputs.

**Comment by egdbear** [comment]
> OK — so I checked this out.
We need some clarification on the behavior here. If I select an end year; presumably the start year should only show years UP TO that year, right?
I think we should actively prevent invalid inputs.

Makes sense!

**Comment by egdbear** [comment]
hey @dzucconi this is ready for 👀

**Review by dzucconi — APPROVED** [review]
👍

**Pull request merged**